src/main/kotlin/org/jitsi/nlj/util/ArrayCache.kt

```diff
@@ -199,6 +207,44 @@ open class ArrayCache<T>(
         }
     }
 
+    /**
+     * Discard an item with a specific index from the cache.
+     */
+    fun discard(index: Int) {
+        if (synchronize) {
+            synchronized(syncRoot) {
+                doDiscard(index)
+            }
+        } else {
+            doDiscard(index)
+        }
+    }
+
+    private fun doDiscard(index: Int) {
+        if (head == -1) {
+            // Not initialized (empty).
+            return
+        }
+
+        val diff = index - cache[head].index
+        if (diff > 0) {
+            // The requested index is newer than the last index we have.
+            return
+        }
+
+        val position = (head + diff) floorMod size
+        if (cache[position].index != index) {
+            return
+        }
+
+        with(cache[position]) {
```

Comment:
I think discarding the last index (with `position == head`) will leave the cache in a bad state.

Reply:
By bad state I mean that `head != -1`, but `cache[head].index == -1`. We can either disallow removing the last item, or loop back to reset `head` to the previous valid entry (or -1)